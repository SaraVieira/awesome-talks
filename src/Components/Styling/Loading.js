import styled, { keyframes } from 'styled-components'
import remcalc from 'remcalc'

const Beat = keyframes`
    0% {
      transform: scale(0.95);
    }
    5% {
      transform: scale(1.1);
    }
    39% {
      transform: scale(0.85);
    }
    45% {
      transform: scale(1);
    }
    60% {
      transform: scale(0.95);
    }
    100% {
      transform: scale(0.9);
    }
`

export default styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translateY(-50%) translateX(-50%) rotate(45deg);
  width: ${remcalc(64)};
  height: ${remcalc(64)};
  transform-origin: ${remcalc(32)} ${remcalc(32)};

  & div {
    top: ${remcalc(23)};
    left: ${remcalc(19)};
    position: absolute;
    width: ${remcalc(26)};
    height: ${remcalc(26)};
    background: #ff0000;
    animation: ${Beat} 1s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
  }

  & div:after,
  & div:before {
    content: ' ';
    position: absolute;
    display: block;
    width: ${remcalc(26)};
    height: ${remcalc(26)};
    background: #ff0000;
  }

  & div:before {
    left: ${remcalc(-17)};
    border-radius: 50% 0 0 50%;
  }

  & div:after {
    top: ${remcalc(-17)};
    border-radius: 50% 50% 0 0;
  }
`
