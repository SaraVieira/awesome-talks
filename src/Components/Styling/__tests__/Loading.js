import React from 'react'
import remcalc from 'remcalc'
import Shallow from '../../../Tests/shallow'
import Loading from '../Loading'

test('Renders Loading', () => {
    const Wrapper = Shallow(<Loading />)

    expect(Wrapper).toHaveStyleRule('width', remcalc(64))
    expect(Wrapper).toHaveStyleRule('height', remcalc(64))
    expect(Wrapper).toMatchSnapshot()
})
