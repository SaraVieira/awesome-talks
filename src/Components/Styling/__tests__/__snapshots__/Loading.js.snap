// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Loading 1`] = `
.c0 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%) rotate(45deg);
  -ms-transform: translateY(-50%) translateX(-50%) rotate(45deg);
  transform: translateY(-50%) translateX(-50%) rotate(45deg);
  width: 4rem;
  height: 4rem;
  -webkit-transform-origin: 2rem 2rem;
  -ms-transform-origin: 2rem 2rem;
  transform-origin: 2rem 2rem;
}

.c0 div {
  top: 1.4375rem;
  left: 1.1875rem;
  position: absolute;
  width: 1.625rem;
  height: 1.625rem;
  background: #ff0000;
  -webkit-animation: gtTijR 1s infinite cubic-bezier(0.215,0.61,0.355,1);
  animation: gtTijR 1s infinite cubic-bezier(0.215,0.61,0.355,1);
}

.c0 div:after,
.c0 div:before {
  content: ' ';
  position: absolute;
  display: block;
  width: 1.625rem;
  height: 1.625rem;
  background: #ff0000;
}

.c0 div:before {
  left: -1.0625rem;
  border-radius: 50% 0 0 50%;
}

.c0 div:after {
  top: -1.0625rem;
  border-radius: 50% 50% 0 0;
}

<div
  className="c0"
/>
`;
